subprojects/gdk_rust/gdk_electrum/src/lib.rs

```diff
@@ -0,0 +1,1212 @@
+        let mut client = ClientWrap::new(self.url.clone())?;
+        let tx_bytes = hex::decode(&tx.hex)?;
+        let txid = client.transaction_broadcast_raw(&tx_bytes)?;
+        Ok(format!("{}", txid))
+    }
+
+    fn broadcast_transaction(&mut self, tx_hex: &str) -> Result<String, Error> {
+        let transaction = BETransaction::from_hex(&tx_hex, self.network.id())?;
+
+        info!("broadcast_transaction {:#?}", transaction.txid());
+        let mut client = ClientWrap::new(self.url.clone())?;
+        let hex = hex::decode(tx_hex)?;
+        let txid = client.transaction_broadcast_raw(&hex)?;
+        Ok(format!("{}", txid))
+    }
+
+    /// The estimates are returned as an array of 25 elements. Each element is
+    /// an integer representing the fee estimate expressed as satoshi per 1000
+    /// bytes. The first element is the minimum relay fee as returned by the
+    /// network, while the remaining elements are the current estimates to use
+    /// for a transaction to confirm from 1 to 24 blocks.
+    fn get_fee_estimates(&mut self) -> Result<Vec<FeeEstimate>, Error> {
+        Ok(self.try_get_fee_estimates().unwrap_or_else(|_| vec![FeeEstimate(1000u64); 25]))
+        //TODO better implement default
+    }
+
+    fn get_mnemonic(&self) -> Result<&Mnemonic, Error> {
+        self.get_wallet().map(|wallet| wallet.get_mnemonic())
+    }
+
+    fn get_settings(&self) -> Result<Settings, Error> {
+        Ok(self.get_wallet()?.get_settings()?)
+    }
+
+    fn change_settings(&mut self, settings: &Settings) -> Result<(), Error> {
+        self.get_wallet()?.change_settings(settings)?;
+        notify_settings(self.notify.clone(), settings);
+        Ok(())
+    }
+
+    fn get_available_currencies(&self) -> Result<Value, Error> {
+        Ok(json!({ "all": [ "USD" ], "per_exchange": { "BITFINEX": [ "USD" ] } }))
+        // TODO implement
+    }
+
+    fn refresh_assets(&self, details: &RefreshAssets) -> Result<Value, Error> {
+        info!("refresh_assets details {:?}", details);
+        let mut map = serde_json::Map::new();
+
+        let wallet = self.get_wallet()?;
+        if details.assets {
+            let assets = wallet
+                .get_asset_registry()?
+                .ok_or_else(|| Error::Generic("cannot find asset registry".into()))?;
+            map.insert("assets".to_string(), assets);
+        }
+        if details.icons {
+            let icons = wallet
+                .get_asset_icons()?
+                .ok_or_else(|| Error::Generic("cannot find asset icons".into()))?;
+            map.insert("icons".to_string(), icons);
+        }
+        Ok(Value::Object(map))
+    }
+
+    fn status(&self) -> Result<u64, Error> {
+        let mut opt = GetTransactionsOpt::default();
+        opt.count = 100;
+        let txs = self.get_wallet()?.list_tx(&opt)?;
+        let mut hasher = DefaultHasher::new();
+        for tx in txs.iter() {
+            std::hash::Hash::hash(&tx.txid, &mut hasher);
```

Comment:
Maybe also include the block height in the hash, similarly to electrum's status_hash?

Reply:
I think it's done 2 lines below

Reply:
I meant the height of each transaction, not of the tip. Though, assuming that transactions and the tip are always updated in sync with each other, the tip should be guaranteed to change whenever the height of any transaction changes, so this shouldn't really matter. But do they necessarily updated in sync?

Reply:
added in afe7eb749b1b2486955e07c870a56bbf361b78ca, required updated in integration tests